File: app/src/main/java/com/bresiu/codechallenge/data/entity/Company.java
package com.bresiu.codechallenge.data.entity;

import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

@Entity(tableName = "companys", foreignKeys = @ForeignKey(entity = User.class, parentColumns = "id", childColumns = "userId", onDelete = ForeignKey.CASCADE))
public class Company {
	@PrimaryKey public int userId;
	public String name;
	public String catchPhrase;
	public String bs;

	public Company(int userId, String name, String catchPhrase, String bs) {
		this.userId = userId;
		this.name = name;
		this.catchPhrase = catchPhrase;
		this.bs = bs;
	}
}

File: app/src/main/java/com/bresiu/codechallenge/data/entity/Company.kt
package com.bresiu.codechallenge.data.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(
    tableName = "companys",
    foreignKeys = [ForeignKey(entity = User::class, parentColumns = ["id"],
        childColumns = ["userId"], onDelete = ForeignKey.CASCADE)]
)
class Company(
    @PrimaryKey var userId: Int,
    var name: String,
    var catchPhrase: String,
    var bs: String
)
